### ISSUE TITLE
`SIGHASH_RANGEPROOF` support

### ISSUE THREAD

**Issue by pavel-kokolemin** [ElementsProject/rust-elements#115]
Is it possible to add `SIGHASH_RANGEPROOF` support?
I implemented something working here - https://github.com/pavel-kokolemin/rust-elements/commit/4cbc317ac320689e7cd214695071cbb1ed10f8c3 but it's not ready for PR (only `SIGHASH_ALL` + `SIGHASH_RANGEPROOF` and without unit tests).

**Comment by apoelstra** [comment]
Yes, we should definitely add this. Thank you for opening the issue.
